erc7824-docs/docs/guides/migration-guide.md

```diff
@@ -0,0 +1,708 @@
+---
+sidebar_position: 2
+title: Migration Guide
+description: Guide to migrate to newer versions of Nitrolite
+keywords: [migration, upgrade, breaking changes, nitrolite, erc7824]
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## 0.3.0 Changelog
+
+This release focuses on streamlining channel creation, improving API consistency, and adding support for modern signature standards.
+
+### Nitrolite SDK
+
+- Replaced an optional NitroliteClient `stateWalletClient` field with a required `StateSigner` interface to handle state signing operations (via `signState(channelId, state)`).
+- Added `WalletStateSigner` (using viem's `WalletClient`) and `SessionStateSigner` (using raw private key account) classes to provide convenient implementations of the `StateSigner` interface.
+- Updated `CreateChannel(...)` method parameters to support the improved channel creation flow.
+- Standardized the request types in all Clearnode API methods, so that now `params` is an object instead of an array. This improves type safety and clarity.
+- Standardized `CreateChannel`, `CloseChannel`, and `ResizeChannel` method response types.
+- Changed the `Signature` type from `{r, s, v}` struct to `Hex`.
+- Added pagination types and parameters for requests to ClearNode pagination-supporting endpoints.
+
+### Clearnode API
+
+- Added `create_channel` method to facilitate the improved, single-transaction channel opening flow. The response includes the `Channel` and `State` structs, alongside with the `serverSignature` for the initial state.
+- Standardized the request structure in all Clearnode API methods, so that now `parameters` is an object instead of an array. This improves type safety and clarity.
```

Comment:
```suggestion
- Standardized the request structure in all Clearnode API methods, so that now `params` is an object instead of an array. This improves type safety and clarity.
```